components/electric/lib/electric/ddlx/parse/common.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Electric.DDLX.Parse.Common do
+  @doc """
+    uses this to template a compiled regex into each parser with the common bits from below and their own keywords
+  """
+  defmacro __using__(_) do
+    quote do
+      @token_re regex_for_keywords(@keywords)
+      @default_schema "public"
+
+      def token_regex() do
+        @token_re
+      end
+
+      def elements() do
+        @elements
+      end
+    end
+  end
+
+  def scope_and_role(role_name) do
+    if String.contains?(role_name, ":") do
+      String.split(role_name, ":")
+      |> List.to_tuple()
+    else
+      {"__global__", role_name}
+    end
+  end
+
+  def schema_and_table(table_name, default_schema) do
+    if String.contains?(table_name, ".") do
+      String.split(table_name, ".")
+      |> List.to_tuple()
+    else
+      {default_schema, table_name}
+    end
+  end
+
+  def expand_privileges(privilege_name) do
+    case privilege_name do
+      "all" -> ["select", "update", "insert", "delete"]
+      "read" -> ["select"]
+      "write" -> ["update", "insert", "delete"]
+      _ -> [privilege_name]
+    end
+  end
+
+  def regex_for_keywords(keywords) do
+    value_types = [
+      ~S"\((?<collection>.+)\)",
+      ~S"'(?<string>[^']+)'",
+      ~S"(?<name>[\p{L}_0-9\$.\/🚀]+)\W"
+    ]
+
+    values_captures = Enum.join(value_types, "|")
+    keyword_captures = Enum.join(keywords, "|")
+    regexstring = "(?i)(?<keyword>#{keyword_captures})|#{values_captures}"
+    {:ok, re} = Regex.compile(regexstring, "u")
+    re
+  end
+
+  def get_value(values, value_name) do
+    case values[value_name] do
+      nil ->
+        nil
+
+      {_, value} ->
+        String.trim(value)
+    end
```

Comment:
Assuming we define `maybe_trim` in this module as
```
defp maybe_trim(nil), do: nil
defp maybe_trim(str), do: String.trim(str)
```

```suggestion
    get_in(values, [value_name, Access.elem(1)]) |> maybe_trim()
```